### PR TITLE
CELDEV-1220 - change URL for new UserAdmin and Ensure loading with correct layout

### PR DESCRIPTION
**Pull request by fpichler**
https://synjira.atlassian.net/browse/CELDEV-1220

### CODE REVIEW

File: src/main/java/com/celements/web/plugin/cmd/CssCommand.java [hunk 1]
```diff
@@ -205,42 +205,33 @@ public List<CSS> includeCSSAfterPageType(String css, XWikiContext context) {
...
-   */
-  @Deprecated
-  private List<BaseObject> addUserSkinCss(Document docAPI) {
-    if ((docAPI != null) && (docAPI.getDocument() != null)) {
-      return addUserSkinCss(docAPI.getDocument());
-    }
-    return Collections.emptyList();
-  }
-
   private List<BaseObject> addUserSkinCss(DocumentReference docRef) {
```
Comment (cbspichler) [inline src/main/java/com/celements/web/plugin/cmd/CssCommand.java:237]: Is this still in use, if support for skins has been dropped?

Comment (fpichler) [reply src/main/java/com/celements/web/plugin/cmd/CssCommand.java:237]: nope

File: src/main/java/com/celements/pagelayout/DefaultLayoutService.java [hunk 2]
```diff
@@ -224,9 +224,20 @@ public String renderPageLayoutLocal(@Nullable SpaceReference layoutSpaceRef) {
     return cellRenderer.getAsString();
   }
 
+  private Optional<SpaceReference> overwriteLayoutRef() {
+    VelocityContext vcontext = ((VelocityContext) modelContext.getXWikiContext().get("vcontext"));
```
Comment (msladek) [inline src/main/java/com/celements/pagelayout/DefaultLayoutService.java]: instead `@Inject` the  `VelocityManager` and use it's `getVelocityContext()`

File: src/main/java/com/celements/pagelayout/DefaultLayoutService.java [hunk 3]
```diff
@@ -224,9 +224,20 @@ public String renderPageLayoutLocal(@Nullable SpaceReference layoutSpaceRef) {
     return cellRenderer.getAsString();
   }
 
+  private Optional<SpaceReference> overwriteLayoutRef() {
+    VelocityContext vcontext = ((VelocityContext) modelContext.getXWikiContext().get("vcontext"));
+    return Optional.ofNullable(vcontext)
+        .map(vc -> vc.get(OVERWRITE_LAYOUT_REF))
+        .filter(SpaceReference.class::isInstance)
+        .map(SpaceReference.class::cast)
```
Comment (msladek) [inline src/main/java/com/celements/pagelayout/DefaultLayoutService.java:234]:
```suggestion
        .flatMap(tryCast(SpaceReference.class))
```
with
```
import static com.celements.common.MoreObjectsCel.*;
```

File: src/main/java/com/celements/web/plugin/cmd/CssCommand.java [hunk 4]
```diff
@@ -284,4 +277,8 @@ private IModelAccessFacade getModelAccess() {
     return Utils.getComponent(IModelAccessFacade.class);
   }
 
+  private LayoutServiceRole getLayoutService() {
+    return Utils.getComponent(LayoutServiceRole.class);
```
Comment (msladek) [inline src/main/java/com/celements/web/plugin/cmd/CssCommand.java]:
```suggestion
    return getSpringContext().getBean(LayoutServiceRole.class);
```
with
```
import static com.celements.spring.context.SpringContextProvider.*;
```

**Review by msladek — CHANGES_REQUESTED** [review]

**Review by cbspichler — COMMENTED** [review]
seems ok to me

**Review by cbspichler — DISMISSED** [review]

File: src/main/java/com/celements/pagelayout/DefaultLayoutService.java (hunk 3 above)
Comment (fpichler) [reply src/main/java/com/celements/pagelayout/DefaultLayoutService.java:234]: does not work, because it is a apply-chain to a optional and not a stream

**Review by msladek — APPROVED** [review]

**Pull request merged**